tests/functional/sync_v3_sign_in.js

```diff
@@ -103,8 +105,40 @@ define([
     },
 
     'unverified': function () {
+      // this test does a lot of waiting around, give it a little extra time
+      this.timeout = 60 * 1000;
+
       return this.remote
-        .then(setupTest(this, false));
+        .then(setupTest(this, false))
+
+        // email 0 - initial sign up email
+        // email 1 - sign in w/ unverified address email
+        // email 2 - "You have verified your Firefox Account"
+
+        // there was a problem with 2 emails being sent on signin,
+        // ensure only one is sent. See #3890. Check for extra email
+        // must be done before opening the verification link,
+        // otherwise the "Account verified!" email is sent.
+
+        // maxAttempts is set to avoid intererence from
+        // the verification reminder emails. 5 attempts occur in 5 seconds,
+        // the first verification reminder is set after 10 seconds.
+        .then(noEmailExpected(email, 2, { maxAttempts: 5 }))
```

Comment:
This is the only place I actually test to ensure the correct emails arrive in the correct order. I only added them here because the test sits around waiting for emails, and it's slow. It already takes ~ 30 mins to run a full test suite. :/